3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/SnowplowAdapter.scala

```diff
@@ -223,9 +230,154 @@ object SnowplowAdapter {
      */
     private def extractAndValidateJson(field: String, schemaCriterion: SchemaCriterion, instance: String)(implicit resolver: Resolver): Validated[JsonNode] =
       for {
-        j <- (JsonUtils.extractJson(field, instance).toValidationNel: Validated[JsonNode])
+        j <- (JU.extractJson(field, instance).toValidationNel: Validated[JsonNode])
         v <- j.verifySchemaAndValidate(schemaCriterion, true).leftMap(_.map(_.toString))
       } yield v
 
   }
+
+  /**
+   * The Redirect Adapter is essentially a pre-processor for
+   * Snowplow Tracker Protocol v2 above (although it doesn't
+   * use the TP2 code above directly).
+   *
+   * The &u= parameter used for a redirect is converted into
+   * a URI Redirect entity and then either stored as an
+   * unstructured event, added to an existing contexts array
+   * or used to initialize a new contexts array.
+   */
+  object Redirect extends Adapter {
```

Comment:
Should SnowplowAdapter become a package rather than an object? Then the various Snowplow Adapters can go into their own files, rather than all in one big file.